hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -699,6 +711,12 @@ public static boolean scmInit(OzoneConfiguration conf,
             + ";cid={};layoutVersion={}", scmStorageConfig.getStorageDir(),
             scmStorageConfig.getClusterID(),
             scmStorageConfig.getLayoutVersion());
+        //TODO: this check need not be done if ratis is not used.
+        // Initialize the raft group Id withe cluster Id.
+        if (isRatisEnabled(conf)) {
+          SCMRatisServerImpl.newRaftServer(clusterId, haDetails,
```

Comment:
Question do we need to initialize RaftServer during this step, as anyway during SCM start, we create RaftServer using SCM ClusterID.

Reply:
Yes, we need to do iniilization.

Reply:
This part of initialization I have not understood why we need during init, as anyway during SCM start we use clusterID and start RaftServer. Can you share more info, why we need this during init step.

Reply:
The idea here, is to just create and initialize the ratis group in init phase. When a ratis server starts up, the group should already exists. Thats way , we differentiate an scm node was initialized and this will immediately start a single node server and will become leader.

The other nodes won't be initialised and after starting up, if figure out they don't belong to any group, first thing to do there is contact the leader by going through each SCM round-robin , initiate an alredy exiting leader to execute setConf to add , it to the ratis ring.